### ISSUE TITLE
how to make different rank use same dataloader in pipeline 

### ISSUE THREAD

**Issue by Stick-To** [microsoft/DeepSpeed#3131]
input and label can not match
when
```
class DatasetDataset(torch.utils.data.Dataset):
    def __init__(self):
        pass

    def __len__(self):
        return 100

    def __getitem__(self, idx):
        n = random.randint(100, 200)
        x = torch.randint(1, 100, (1, n)).to(torch.long)
        x = torch.stack((x, torch.ones_like(x)), dim=1)
        y = torch.zeros(1, n).to(torch.long)
        return x, y

dataset = DatasetDataset() 
engine, optim, _, _ = deepspeed.initialize(
        args=args,
        model=model,
        model_parameters=model.parameters(),
        training_data=dataset,
    ) 

for i in range(10000): 
     loss = engine.train_batch() 
     print(loss)
```

**Comment by Stick-To** [comment]
does the input of different batch if pipeline model need have same size?
